website/docs/generated/datasources/transform/decode/role_name.md

```diff
@@ -0,0 +1,52 @@
+---
```

Comment:
I feel like these should be moved out of the `generated/` folder. The URL will have this path and the docs are currently generate-once with manual manipulation after that. What do you think?

Reply:
Oh - the docs aren't generate-once. There are two main reasons it's like this:

1. To flag to folks that these are generated files and shouldn't be manually edited, because there are a lot of community contributions to the Terraform Vault Provider.
2. Each time a new OpenAPI doc is added and `$ make generate` is run, if an endpoint has changed, the new file will overwrite the previous one. Then there'll be a visible diff - a parameter appeared or disappeared! Then we can both add immediate support for the current state, and it flags us to begin [this process](https://www.terraform.io/docs/extend/best-practices/deprecations.html). 

On handling deprecations, renames, removals, etc., I think where the file should live during that process is currently TBD but can be determined as needed moving forward. One option would be it's left in place, and manually edited, but future additional changes are easy to see from the endpoint. Another option is to move it into the `vault` folder for manual upkeep, which might be nicer, but would have the downside of not highlighting changes.

Reply:
I'm confused - isn't a given doc file protected from being generated a second time if one already exists? https://github.com/terraform-providers/terraform-provider-vault/pull/749/files#diff-7b9765054e16f7d1b8cc96e6fca7e858R94-R95